addons/mass_mailing/models/mass_mailing.py

```diff
@@ -80,92 +63,75 @@ class MassMailingContact(osv.Model):
     _order = 'email'
     _rec_name = 'email'
 
-    _columns = {
-        'name': fields.char('Name'),
-        'email': fields.char('Email', required=True),
-        'create_date': fields.datetime('Create Date'),
-        'list_id': fields.many2one(
-            'mail.mass_mailing.list', string='Mailing List',
-            ondelete='cascade', required=True,
-        ),
-        'opt_out': fields.boolean('Opt Out', help='The contact has chosen not to receive mails anymore from this list'),
-        'unsubscription_date': fields.datetime('Unsubscription Date'),
-        'message_bounce': fields.integer('Bounce', help='Counter of the number of bounced emails for this contact.'),
-    }
-
-    def _get_latest_list(self, cr, uid, context={}):
-        lid = self.pool.get('mail.mass_mailing.list').search(cr, uid, [], limit=1, order='id desc', context=context)
-        return lid and lid[0] or False
-
-    _defaults = {
-        'list_id': _get_latest_list
-    }
-
-    def on_change_opt_out(self, cr, uid, id, opt_out, context=None):
-        return {'value': {
-            'unsubscription_date': opt_out and fields.datetime.now() or False,
-        }}
-
-    def create(self, cr, uid, vals, context=None):
+    name = fields.Char()
+    email = fields.Char(required=True)
+    create_date = fields.Datetime(string='Create Date')
+    list_id = fields.Many2one(
+        'mail.mass_mailing.list', string='Mailing List',
+        ondelete='cascade', required=True, default=lambda self: self.env['mail.mass_mailing.list'].search([], limit=1, order='id desc'))
+    opt_out = fields.Boolean(string='Opt Out', help='The contact has chosen not to receive mails anymore from this list')
+    unsubscription_date = fields.Datetime(string='Unsubscription Date')
+    message_bounce = fields.Integer(string='Bounce', help='Counter of the number of bounced emails for this contact.')
+
+    @api.onchange('opt_out')
+    def on_change_opt_out(self):
+        self.unsubscription_date = self.opt_out and fields.Datetime.now()
+
+    @api.model
+    def create(self, vals):
         if 'opt_out' in vals:
-            vals['unsubscription_date'] = vals['opt_out'] and fields.datetime.now() or False
-        return super(MassMailingContact, self).create(cr, uid, vals, context=context)
+            vals['unsubscription_date'] = vals['opt_out'] and fields.Datetime.now() or False
```

Comment:
@kga-odoo still you can avoid this overriding methods ```create() and write()``` by changing your code in ```on_change_opt_out()``` like
```
self.unsubscription_date = self.opt_out and fields.Datetime.now() or ''
```

if you you have demo data with ```opt_out``` field ```True``` value then you can change this demo data and add ```unsubscription_date``` field in this demo data